### ISSUE TITLE
textures

### ISSUE THREAD

**Issue by weepy** [patriciogonzalezvivo/glslEditor#29]
Is it possible to import image textures ?

**Comment by ChronoAndross** [comment]
Well from the looks of it, it looks like texture2D is supported within the editor, but currently can't be used for anything since there is no option to import image textures. I think on the main page of glslEditor, there is a request to create a texture editor, so I can look into this from my end and see what I can do.

**Comment by weepy** [comment]
that'd be awesome.

**Comment by patriciogonzalezvivo** [comment]
Hi! GlslEditor supports textures... you just have to write the URL as a comment next to texture uniform. http://editor.thebookofshaders.com/?log=180130173243

Also watch out for CORS!

**Comment by weepy** [comment]
Oh sweet !

**Comment by weepy** [comment]
Is video possible as a texture ?

**Comment by patriciogonzalezvivo** [comment]
@weepy yes... any format that the browser can transform into a texture is supported
@ChronoAndross sounds great! Would like to sync on the UI when you have a moment. How do you imagine? At the end of the day should be pass as a comment next the uniforms. So the glslCanvas can open it even in "player" mode.

**Comment by ChronoAndross** [comment]
@patriciogonzalezvivo I can post the UI when I get home (just leaving work now) but yes, if the file can be opened successfully, it should be placed as a comment next to the uniform named "utex_(#texture)", the newly created texture based on the user's imported image.

And then your code will run as usual.

**Comment by patriciogonzalezvivo** [comment]
sounds great

**Comment by ChronoAndross** [comment]
@patriciogonzalezvivo 

Here's what the new page will look like with my new code:
![screen shot 2018-02-03 at 4 46 05 pm](https://user-images.githubusercontent.com/3382216/35771966-78de004e-0903-11e8-8f85-74f7a9527202.png)

Notice the new "Import" button at the top there. It contains a drop down which contains 2 new options: "Import Image URL..." and "Import Local Image..." After pressing "Import Local Image...", the user will see the following:
![screen shot 2018-02-03 at 4 46 58 pm](https://user-images.githubusercontent.com/3382216/35771960-67a9da3c-0903-11e8-9f92-48a9c8a1c519.png)


After the user selects an image, it will load it into the current shader with the name "u_tex" + the number of the active texture. The first texture that is loaded should be 0, so textures should be numbered from 0 -> n number of textures that are loaded.

If it notices "u_tex + the number of the active texture" is already in the shader, it will replace the comment that is already there for convenience. Here what my image looked like after I loaded it with my option:
![screen shot 2018-02-03 at 4 46 25 pm](https://user-images.githubusercontent.com/3382216/35771958-5ab319f6-0903-11e8-94a6-6f5d2fa10909.png)


When loading an image via a URL, the user is prompted to insert an image:
![screen shot 2018-02-03 at 4 46 43 pm](https://user-images.githubusercontent.com/3382216/35771954-540508a8-0903-11e8-9f41-bea23f1d8718.png)

There's one problem with it I'm seeing thus far: I'm currently not catching the CORS failures if they happen since they get rejected way far down into the code. I can probably fix this bug if you want me to. But otherwise, I think it's ready. 

Do you want me to submit a pull request for my changes to prepare to merge?

Thanks!

**Comment by felixfaire** [comment]
Did any of this texture loading work make it's way into the web version of glslEditor hosted at http://editor.thebookofshaders.com/ ? I'm struggling to replicate this feature and would love to use it for prototyping and also running a workshop (where having students use the web version is a lot easier than making people download native tools). Thanks.
